src/components/TodoList/TodoList.js

```diff
@@ -1,41 +1,69 @@
 import React from 'react';
 import './TodoList.css';
 
-function handleClick() {
-  console.log(this)
-}
-
 class TodoList extends React.Component {
 
-  // state = {
-  //   list: [
-  //     { id: 1, name: 'Fumer des cannes', active: true },
-  //     { id: 2, name: 'Coder cette shit', active: true },
-  //     { id: 3, name: 'Nager avec des baleines', active: true },
-  //     { id: 4, name: 'Rouler des cannes pour demain', active: true }
-  //   ],
-  //   newTaskName: "",
-  // // }
-  //
-  // handleClick = () => {
-  //   console.log(this)
-  // }
+  state = {
+    list: [ ],
+    newTaskName: "",
+    editTaskName: ""
+  }
 
   deleteTodo(itemToDeleteId) {
     const newList = this.state.list.filter((item) => item.id !== itemToDeleteId)
     this.setState({ list: newList })
   }
 
   createNewTask() {
+    if (this.state.list.length === 0) {
+      this.setState({
+        list: [{ id: 1, name: this.state.newTaskName, active: true, editing: false }],
+        newTaskName: ""
+      })
+      return;
+    }
+
     this.setState({
       list: [
         ...this.state.list,
-        { id: this.state.list[this.state.list.length - 1].id + 1, name: this.state.newTaskName, active: true }
+        { id: this.state.list[this.state.list.length - 1].id + 1, name: this.state.newTaskName, active: true, editing: false}
       ],
-      newTaskName: ""
+      newTaskName: "",
+      editTaskName: ""
     })
   }
 
+  displayEdit(taskId) {
+    const { list } = this.state;
+    const taskIndex = list.findIndex(task => task.id == taskId)
+    const task = list[taskIndex]
+
+    this.setState({ list: [
+      ...list.slice(0, taskIndex),
+    {
+      id: task.id,
+      name: task.name,
+      active: task.active,
+      editing: !task.editing
+    },
+    ...list.slice(taskIndex + 1, list.length)
+    ]})
+  }
+
+  editTask(taskId) {
+    const { list } = this.state;
+    const taskIndex = list.findIndex(task => task.id == taskId)
+    const task = list[taskIndex]
+
+    this.setState({ list: [
+      ...list.slice(0, taskIndex),
+    { id: this.state.list[this.state.list.length - 1].id + 1, name: this.state.editTaskName, active: true, editing: false },
```

Comment:
I think you have an error here.
`editTask` should have the following effect : 

1. switch `editing` off
2. set the name to `editTaskName`
 
This could be expressed by the following code: `{ ...task, editing: false, name: editTaskName }`

But in your code, you also modify the id, for no reason, which leads to unexpected bugs:
<img width="480" alt="Capture d’écran 2020-07-16 à 13 41 21" src="https://user-images.githubusercontent.com/7590272/87667194-3487d780-c76a-11ea-92bb-f1e66d05ba01.png">
Take a look at the different ids